### ISSUE TITLE
reproducing the results

### ISSUE THREAD

**Issue by msed-Ebrahimi** [liuf1990/CFSM#3]
Thanks for the great work and comprehensive repository.
I tried to reproduce the results (FR part). Still, after training (FR backbone is being trained from scratch, and for the generator, I used the pre-trained weights of WiderFace from the repository), the results on Tinyface are lower than what is reported on the paper and also lower than the performance of the pre-trained R50 from the repository?
Am I missing something? 

I appreciate any help you can provide.

**Comment by liufeng2915** [comment]
Hi, 
Would you please elaborate on your training process, so that I might be able to locate the problem? E.g.,
1. Were you using the pre-trained synthesis model that was trained on 70K WiderFaceimages? The Pre-trained weights are available at https://drive.google.com/file/d/1dm2IBJ5L158duI8DczBFn6SuqZ0xnekb/view?usp=share_link.
2. Did you train your face recognition model on the MS1M-V2 dataset?
Thank you!

**Comment by msed-Ebrahimi** [comment]
Hi, 
Thanks for your prompt response.
1- Yes I did use the pre-trained weights from the link.
2- Yes.
here is the logfile for my training(Everything is like the repository).

Training: 2022-12-22 12:36:41,757-rank_id: 0
Training: 2022-12-22 12:36:48,889-: loss                     arcface
Training: 2022-12-22 12:36:48,889-: output                   checkpoints/arcface_ms1mv2_r50_ours
Training: 2022-12-22 12:36:48,889-: network                  r50
Training: 2022-12-22 12:36:48,889-: resume                   False
Training: 2022-12-22 12:36:48,889-: pretrained_backbone_modelNone
Training: 2022-12-22 12:36:48,889-: pretrained_partial_fc_pathNone
Training: 2022-12-22 12:36:48,889-: embedding_size           512
Training: 2022-12-22 12:36:48,890-: sample_rate              1.0
Training: 2022-12-22 12:36:48,890-: fp16                     True
Training: 2022-12-22 12:36:48,890-: momentum                 0.9
Training: 2022-12-22 12:36:48,890-: weight_decay             0.0005
Training: 2022-12-22 12:36:48,890-: batch_size               256
Training: 2022-12-22 12:36:48,890-: lr                       0.1
Training: 2022-12-22 12:36:48,890-: verbose                  2000
Training: 2022-12-22 12:36:48,890-: dali                     False
Training: 2022-12-22 12:36:48,890-: frequent                 10
Training: 2022-12-22 12:36:48,890-: score                    None
Training: 2022-12-22 12:36:48,890-: rec                      FaceRecDatasets/faces_ms1m_112x112/
Training: 2022-12-22 12:36:48,890-: num_classes              85742
Training: 2022-12-22 12:36:48,890-: num_image                3870313
Training: 2022-12-22 12:36:48,890-: num_epoch                20
Training: 2022-12-22 12:36:48,890-: warmup_epoch             2
Training: 2022-12-22 12:36:48,890-: val_targets              ['lfw']
Training: 2022-12-22 12:36:48,890-: rs_ratio                 0.75
Training: 2022-12-22 12:36:48,890-: pretrained_synthesis_modelcfsm_checkpoint/WiderFace.pth
Training: 2022-12-22 12:36:48,890-: epsilon                  0.314
Training: 2022-12-22 12:36:48,890-: alpha                    0.314
Training: 2022-12-22 12:36:48,890-: k                        1
Training: 2022-12-22 12:36:48,890-: warmup_step              15118

**Comment by liufeng2915** [comment]
1. Would you please use WiderFace70K.pth as the synthesis model and retrain your face recognition models?
2.  Please be aware that if you used MS1MV2 (MS1M-ArcFace from [ArcFace](https://github.com/deepinsight/insightface/tree/master/recognition/_datasets_)), you need to set num_image=<del>3870313 </del>5822653

**Comment by msed-Ebrahimi** [comment]
Thanks for your clarification.
Yes, I would.
I think I might have been mistaken.
So the checkpoint from "[ArcFace](https://github.com/deepinsight/insightface/tree/master/recognition/arcface_torch)+CFSM" is trained on [MS1M-ArcFace (85K ids/5.8M images) ](https://drive.google.com/file/d/1SXS4-Am3bsKSK615qbYdbA_FMVh3sAvR/view?usp=sharing) not [MS1M-IBUG (85K ids/3.8M images)](https://pan.baidu.com/s/1nxmSCch). right?

**Comment by liufeng2915** [comment]
Yes. We used MS1MV2-clean which is a subset of MS1M-ArcFace.

**Comment by msed-Ebrahimi** [comment]
Is it publicly available? In the beginning I thought MS1MV2-clean is the same as [MS1M-IBUG (85K ids/3.8M images) ](https://pan.baidu.com/s/1nxmSCch)

**Comment by liufeng2915** [comment]
Please find the MS1MV2-clean data [here](https://drive.google.com/file/d/1unzZRk0Na_Z5jkxefxbyf2DVV3iHD_oB/view?usp=share_link).

**Comment by msed-Ebrahimi** [comment]
Thank you very much

**Comment by msed-Ebrahimi** [comment]
> Please find the MS1MV2-clean data [here](https://drive.google.com/file/d/1unzZRk0Na_Z5jkxefxbyf2DVV3iHD_oB/view?usp=share_link).
Sorry for the trouble. 
The train.idx file is missing.

**Comment by Edgar-Lzxc** [comment]
> Please find the MS1MV2-clean data [here](https://drive.google.com/file/d/1unzZRk0Na_Z5jkxefxbyf2DVV3iHD_oB/view?usp=share_link).

I want to use adaface loss function to train adaface dataset, is there still an error reported that prevents training?

**Comment by bodhuehiz** [comment]
> Please find the MS1MV2-clean data [here](https://drive.google.com/file/d/1unzZRk0Na_Z5jkxefxbyf2DVV3iHD_oB/view?usp=share_link).

Hi @liufeng2915 Could you also provide the `train.idx` file or the cleaning script you use on MS1MV2?

Thanks!

**Comment by liufeng2915** [comment]
Please find the train.idx  [here](https://drive.google.com/drive/folders/1olfwDOqc04IgneIH_wNklkJv5SiQvU5E?usp=sharing).
